docs/source/docker/advanced/datasource_metadata.rst

```diff
@@ -0,0 +1,177 @@
+of the images we have collected. A possible schema could look like this:
+
+.. code-block:: javascript
+    :caption: .lightly/metadata/schema.json
+
+    [
+        {
+            "name": "Is special frame",
+            "path": "special_frame_flag",
+            "defaultValue": false,
+            "valueDataType": "CATEGORICAL_BOOLEAN"
+        },
+        {
+            "name": "Weather description",
+            "path": "weather.description",
+            "defaultValue": "nothing",
+            "valueDataType": "CATEGORICAL_STRING"
+        },
+        {
+            "name": "Temperature",
+            "path": "weather.temperature",
+            "defaultValue": 0.0,
+            "valueDataType": "NUMERIC_FLOAT"
+        },
+        {
+            "name": "Air pressure",
+            "path": "weather.air_pressure",
+            "defaultValue": 0,
+            "valueDataType": "NUMERIC_INT"
+        },
+        {
+            "name": "Vehicle ID",
+            "path": "vehicle_id",
+            "defaultValue": 0,
+            "valueDataType": "CATEGORICAL_INT"
+        }
+    ]
+
+
+
+
+Metadata Files
+--------------
+Lightly requires a single metadata file per image or video. If a metadata file is provided
+for a full video, Lightly assumes that the metadata is valid for all frames in that video.
+
+To provide metadata for an image or a video, place a metadata file with the same name
+as the image or video in the `.lightly/metadata` directory but change the file type to
+`.json`. The file should contain the metadata in the format defined under :ref:`ref-metadata-format`.
+
+
+.. code-block:: bash
+
+    # filename of the metadata for file FILENAME.EXT
+    .lightly/metadata/${FILENAME}.json
+
+    # example: my_image.png
+    .lightly/metadata/my_image.json
+
+    # example: my_video.mp4
+    .lightly/metadata/my_video.json
+
+
+When working with videos it's also possible to provide metadata on a per-frame basis.
+Then, Lightly requires a metadata file per frame. Lightly uses a naming convention to
+identify frames: The filename of a frame consists of the video filename, the video format,
+and the frame number (padded to the length of the number of frames in the video) separated
+by hyphens. For example, for a video with 200 frames, the frame number will be padded
+to length three. For a video with 1000 frames, the frame number will be padded to length four (99 becomes 0099).
+
+
+.. code-block:: bash
+
+    # filename of the metadata of the Xth frame of video FILENAME.EXT
+    # with 200 frames (padding: len(str(200)) = 3)
+    .lightly/metadata/${FILENAME}-${X:03d}-${EXT}.json
+
+    # example: my_video.mp4, frame 99/200
+    .lightly/metadata/my_video-099-mp4.json
+
+    # example: my_subdir/my_video.mp4, frame 99/200
+    .lightly/metadata/my_subdir/my_video-099-mp4.json
+
+
+.. _ref-metadata-format:
+
+Metadata Format
+---------------
+
+Metadata for images or videos must have a `file_name`, `type`, and `metadata`` key.
```

Comment:
suggestion: Metadata for images or videos must have a -> The metadata `json` files for images and videos require the keys